pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -229,6 +255,266 @@ func (p *Pod) Skip() bool {
 	return false
 }
 
+func (p *Pod) groupName() string {
+	return p.Object().GetLabels()[GroupNameLabel]
+}
+
+func (p *Pod) groupTotalCount() (int, error) {
+	if p.groupName() == "" {
+		return 0, fmt.Errorf("pod doesn't have a '%s' label", GroupNameLabel)
+	}
+
+	gtcAnnotation, ok := p.Object().GetAnnotations()[GroupTotalCountAnnotation]
```

Comment:
Let's assume 1 by default. Some users might want to use a group of 1 Pod to be able to retry failed pods without having to queue a new one.

Reply:
The validating webhook will reject a pod with the group name but without the group total count. 

I would rather keep the same behavior and let the user decide on the pod group size.